woocommerce/woocommerce-ios
# Order Creation: Integrate remote order creation action in UI layer

Closes: #5405, #5457

## Description

This PR integrates the `createOrder` action from the Yosemite layer into the New Order view, in the order creation flow. The New Order view model now displays the Create button when any order details have changed from their default values. When the network request is being performed, a loading indicator is displayed. And when the network request is complete, an error notice is displayed if the order creation fails.

## Changes

* Updates `NewOrder` to display the expected navigation bar trailing item, based on the `NavigationItem` cases.
* Changes to `NewOrderViewModel`:
   * Adds an initializer for the view model.
   * Adds a property to define what navigation bar trailing button to display, based on the `NavigationItem` enum. This defaults to no button (to match the designs), changes to the Create button when the order details have changed from their default values, and changes to a loading indicator when the network request is being made.
   * Adds a property to hold the order details, based on the `OrderDetails` struct. For now this type includes the details we will be using for the first milestone of order creation (status, products, customer details).
   * Adds a method to make the network request to create the order.
* Adds unit tests for `NewOrderViewModel`.

## Testing

We don't have anything in the UI yet to update the order details. To test this you can temporarily add something to the `NewOrder` view that will update the order details, e.g. changing the ScrollView to this:
```
        ScrollView {
            Button("Processing") {
                viewModel.orderDetails.status = .processing
            }
        }
```

To test:

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button.
4. Confirm there is no "Create" button by default.
4. Tap the temporary "Processing" button to change the order status and confirm the Create button appears.
5. Tap the "Create" button and confirm the button changes to a loading indicator while the network request is happening.
6. Confirm the "Create" button reappears when the network request completes.
7. Disable your internet connection, tap the "Create" button again, and confirm an error notice appears.

## Submitter Checklist

Update release notes:
